### PR TITLE
[mysql-connector-cpp] upgrade to 8.0.32

### PR DESCRIPTION
**Pull request by kotori2**
I did these things:
* Upgrade to 8.0.32
* Use all dependencies from vcpkg
* Include `jdbc` support (Windows dynamic linkage is not supported, because it will require libmysql:windows-xx-static and link them into a big dll. I didn't find a way to achieve that in vcpkg.)

Checklist: 
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/mysql-connector-cpp/vcpkg.json [hunk 1]
```diff
@@ -1,20 +1,33 @@
 {
   "name": "mysql-connector-cpp",
-  "version": "8.0.30",
+  "version": "8.0.32",
```
Comment (jimwang118) [inline ports/mysql-connector-cpp/vcpkg.json:3]:
```suggestion
  "version": "8.0.33",
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: ports/mysql-connector-cpp/vcpkg.json (hunk 1 above)
Comment (kotori2) [reply ports/mysql-connector-cpp/vcpkg.json:3]: I'm wondering... Why it should be `8.0.33`?

Comment (jimwang118) [reply ports/mysql-connector-cpp/vcpkg.json:3]: > I'm wondering... Why it should be `8.0.33`?

I just checked that the latest upstream version is 8.0.33. It is recommended to update to the latest version.

Comment (kotori2) [reply ports/mysql-connector-cpp/vcpkg.json:3]: It's a little bit weird that [mysql-server](https://github.com/mysql/mysql-server/tags) is still at `8.0.32`... I guess the version of connector should match the server?

Comment (jimwang118) [reply ports/mysql-connector-cpp/vcpkg.json:3]: > It's a little bit weird that [mysql-server](https://github.com/mysql/mysql-server/tags) is still at `8.0.32`... I guess the version of connector should match the server?

Ok, then there should be no problem with this one, no modification needed.

**Review by jimwang118 — DISMISSED** [review]

**Comment by jimwang118** [comment]
Have you tested its usage after installation? I followed the usage prompt after installation: 
```
find_package(unofficial-mysql-connector-cpp CONFIG REQUIRED)
target_link_libraries(main PRIVATE unofficial::mysql-connector-cpp::connector)
```
the header files in `installed/x64-windows/include` in my test project cannot be referenced in the project.

**Comment by kotori2** [comment]
> Have you tested its usage after installation? I followed the usage prompt after installation:
> 
> ```
> find_package(unofficial-mysql-connector-cpp CONFIG REQUIRED)
> target_link_libraries(main PRIVATE unofficial::mysql-connector-cpp::connector)
> ```
> 
> the header files in `installed/x64-windows/include` in my test project cannot be referenced in the project.

It works for me. Could you upload your test case? 
image

**Comment by jimwang118** [comment]
CMakeLists.txt
```
add_executable (testmysqlcon "testmysqlcon.cpp" "testmysqlcon.h")
find_package(unofficial-mysql-connector-cpp CONFIG REQUIRED)
target_link_libraries(testmysqlcon PRIVATE unofficial::mysql-connector-cpp::connector)
if (CMAKE_VERSION VERSION_GREATER 3.12)
  set_property(TARGET testmysqlcon PROPERTY CXX_STANDARD 20)
endif()
```
code:
```
#include "testmysqlcon.h"
#include <mysqlx/common.h>
using namespace std;

int main()
{
	cout << "Hello CMake." << endl;
	return 0;
}
```
error:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C1083	Cannot open include file: 'mysqlx/common.h': No such file or directory	testmysqlcon	F:\testwork\testmysqlcon\testmysqlcon\testmysqlcon.cpp	5	
```

**Comment by jimwang118** [comment]
@kotori2 Is my usage wrong? Can you tell me how you use it?

**Comment by kotori2** [comment]
> CMakeLists.txt
> 
> ```
> add_executable (testmysqlcon "testmysqlcon.cpp" "testmysqlcon.h")
> find_package(unofficial-mysql-connector-cpp CONFIG REQUIRED)
> target_link_libraries(testmysqlcon PRIVATE unofficial::mysql-connector-cpp::connector)
> if (CMAKE_VERSION VERSION_GREATER 3.12)
>   set_property(TARGET testmysqlcon PROPERTY CXX_STANDARD 20)
> endif()
> ```
> 
> code:
> 
> ```
> #include "testmysqlcon.h"
> #include <mysqlx/common.h>
> using namespace std;
> 
> int main()
> {
> 	cout << "Hello CMake." << endl;
> 	return 0;
> }
> ```
> 
> error:
> 
> ```
> Severity	Code	Description	Project	File	Line	Suppression State
> Error	C1083	Cannot open include file: 'mysqlx/common.h': No such file or directory	testmysqlcon	F:\testwork\testmysqlcon\testmysqlcon\testmysqlcon.cpp	5	
> ```

It seems that my test case works because I included other vcpkg packages to my build, so mysql headers get included as well. 
I tried the previous version with commit hash 2a6dfa3a7e2e926aacef5402b3c26f7f433052b7, it turns out it does not work as well. Gonna test if it is a regression issue.

**Comment by kotori2** [comment]
I tracked back to the very first version 79211d3e4a8c79bc81858f1622b81d6ca6fed72e and header will not include on their on, and I have no idea how to fix it. 
Should I add `include_directories` to `mysql-connector-cpp-config.cmake.in`?

**Comment by dg0yt** [comment]
You need to extend this patch to provide a public install interface:
https://github.com/microsoft/vcpkg/blob/2b1757dd1d5ed7d9d5bb65fd8316631b157e7afc/ports/mysql-connector-cpp/export-targets.patch#L9-L10

It is missing from many vcpkg patches. Example from another PR (may need adjustments):
https://github.com/microsoft/vcpkg/blob/ff80c0832cbcc2245db1bb3ce838b722d58e90d2/ports/minizip/CMakeLists.txt#L27

As an extra note, IMO the port deserves more work. For example it should really use a standard `find_package(ZLIB REQUIRED)` instead of custom zlib lookup code.

**Comment by kotori2** [comment]
> For example it should really use a standard `find_package(ZLIB REQUIRED)` instead of custom zlib lookup code.

Oh okay. I was trying to touch as fewer code as possible. Gonna try these later when I have time.

**Comment by dg0yt** [comment]
> Oh okay. I was trying to touch as fewer code as possible. Gonna try these later when I have time.

Your intentions are good :+1: 
But it is more maintainable to solve the cross-platform issues once in a single spot, e.g. within port zlib, than in all consuming ports. And maybe you may find a minimal change by injecting the `find_package(ZLIB)` early enough in the call stack. (I don't know if it is feasible here.)

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by dg0yt** [comment]
Oh, they really circumvent canonical `find_package(...)` calls even for zlib and openssl? 
If not restoring `find_package`, this port will be hard to maintain and use: duplicating what we already maintain in wrappers and still not delivering imported targets which CMake can identify with canonical targets such as `ZLIB::ZLIB` or `OpenSSL::SSL`.

**Comment by kotori2** [comment]
> Oh, they really circumvent canonical `find_package(...)` calls even for zlib and openssl? If not restoring `find_package`, this port will be hard to maintain and use: duplicating what we already maintain in wrappers and still not delivering imported targets which CMake can identify with canonical targets such as `ZLIB::ZLIB` or `OpenSSL::SSL`.

@dg0yt I do tried on my end with such patch:
```diff
diff --git a/cdk/cmake/DepFindCompression.cmake b/cdk/cmake/DepFindCompression.cmake
index f9fe519f..ed256026 100644
--- a/cdk/cmake/DepFindCompression.cmake
+++ b/cdk/cmake/DepFindCompression.cmake
@@ -44,6 +44,10 @@ if(TARGET ext::z)
 endif()

 message(STATUS "Setting up compression libraries.")
+find_package(ZLIB REQUIRED)
+find_package(lz4 CONFIG REQUIRED)
+find_package(zstd REQUIRED)
+return()

 #######
 # ZLIB
```

But I got an weird error:
```
-- Setting up compression libraries.
CMake Error at C:/vcpkg/downloads/tools/cmake-3.25.1-windows/cmake-3.25.1-windows-i386/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
Call Stack (most recent call first):
  C:/vcpkg/downloads/tools/cmake-3.25.1-windows/cmake-3.25.1-windows-i386/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/vcpkg/downloads/tools/cmake-3.25.1-windows/cmake-3.25.1-windows-i386/share/cmake-3.25/Modules/FindZLIB.cmake:166 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cdk/cmake/DepFindCompression.cmake:47 (find_package)
  cdk/cmake/dependency.cmake:45 (include)
  cdk/protocol/mysqlx/CMakeLists.txt:30 (find_dependency)
```
I am pretty sure ZLIB is required in `vcpkg.json`. Could you please help me out?



Also the CI seems failing randomly on Windows
```
========================== Starting Command Output ===========================
"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "C:\a\_temp\a25dcb94-6018-49c8-a186-ac64e33887bb.cmd""
Succeeded in formatting the manifest files.
vcpkg has crashed. Please create an issue at *** containing a brief summary of what you were trying to do and the following information.

Version=2023-06-08-19a3cb921a585fce63c0651f56a991a5a08bdc60
EXCEPTION=bad allocation
CMD=
.\vcpkg.exe|
format-manifest|
--all|
##[error]Cmd.exe exited with code '1'.
Finishing: Format Manifests
```

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

**Comment by kotori2** [comment]
I'm sorry but I think changing all package search calls from `add_ext` to `find_package` is out of my capability because the code are not only searching for library/headers, but also binaries such as protobuf. I did tried but it doesn't work. 
We might need a separate PR to enhance this issue since this PR is already delayed for 2 months. 
Again thanks for the help from @dg0yt :D

**Pull request merged**